engines/postgres.py

```diff
@@ -122,6 +124,28 @@ def format_insert_value(self, value, datatype):
                 pass
         return Engine.format_insert_value(self, value, datatype)
 
+    def to_csv(self):
+        tablename = str(Engine.table_name(self))
+        tablefile = str(self.table.name)
+        csvfile_output = os.path.normpath(str(tablefile + '.csv'))
+        self.get_cursor()
+        sql_query = ("SELECT * FROM " + tablename + ";")
+        self.cursor.execute(sql_query)
+
+        row = self.cursor.fetchone()
+        colnames = [tuple_i[0] for tuple_i in self.cursor.description]
+        csv_out = open(csvfile_output, "wb")
+        csv_writer = csv.writer(csv_out, dialect='excel')
+        csv_writer.writerow(colnames)
+
+        while row is not None:
+            csv_writer.writerow([values for values in row])
```

Comment:
very good question :+1: 
So basically most of these `str() ` problems in conjunction with list compression where initially introduced as I was debugging. At some point I had to pre-process each value.